### PR TITLE
RTT Disconnect addition BCLD-6256

### PR DESCRIPTION
**Pull request by pascalf1309**

### CODE REVIEW

**Comment by Daivuk** [comment]
Can you put the JIRA # in the title. I am unsure what this relates to

File: src/RTTComms.cpp [hunk 1]
```diff
@@ -109,6 +109,12 @@ namespace BrainCloud
 
             delete _socket;
             _socket = NULL;
+            if(_disconnectedWithReason==true)
```
Comment (davidstl) [inline src/RTTComms.cpp:112]: 🚲 
`if (_disconnectedWithReason == true)`

File: src/RTTComms.cpp [hunk 2]
```diff
@@ -543,16 +550,28 @@ namespace BrainCloud
...
+                startHeartbeat();
 
-            _eventQueueMutex.lock();
-            _callbackEventQueue.push_back(RTTCallback(RTTCallbackType::ConnectSuccess));
-            _eventQueueMutex.unlock();
+                _eventQueueMutex.lock();
+                _callbackEventQueue.push_back(RTTCallback(RTTCallbackType::ConnectSuccess));
+                _eventQueueMutex.unlock();
+            }
+            if (operation == "DISCONNECT")
```
Comment (davidstl) [inline src/RTTComms.cpp:566]: `else if`
We don't want to be comparing another string if first `if` passed.

File: src/RTTComms.cpp [hunk 3]
```diff
@@ -543,16 +550,28 @@ namespace BrainCloud
...
+            {
+                _disconnectedWithReason = true;
+                _disconnectReasonMessage = json["data"]["reason"].asString();
+                _disconnectReasonCode = json["data"]["reasonCode"].asInt();
+
+			    _msg["reasonCode"] = _disconnectReasonCode;
```
Comment (davidstl) [inline src/RTTComms.cpp:572]: 🚲 
Replace tabs with spaces

File: src/RTTComms.cpp [hunk 4]
```diff
@@ -109,6 +109,12 @@ namespace BrainCloud
 
             delete _socket;
             _socket = NULL;
+            if(_disconnectedWithReason==true)
+            {
+                Json::FastWriter myWriter;
+                std::string output ="\n" + myWriter.write(_msg); 
+                printf(output.c_str());
```
Comment (davidstl) [inline src/RTTComms.cpp:116]: I think we wanted to do more than just print it out? I don't remember. Don't want this to be sent to _connectCallback failure?

But at least the user will have log. So leave it like that, we're good. This is a very rare case anyway.

**Review by davidstl — CHANGES_REQUESTED** [review]
Looks good. Minor changes 🚲 required.
I presume this is not tested? I think it's very hard to repro case. So as long as other tests are still passing and C++ apps.

**Pull request merged**